### PR TITLE
[NEAT-111] DMS Rules

### PR DESCRIPTION
**Pull request by doctrino**

### CODE REVIEW

File: cognite/neat/rules/models/_rules/dms_architect_rules.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    except AttributeError:
+        ...
+del subclasses  # cleanup namespace
+
+
+class DMSArchitectMetadata(BaseMetadata):
+    role: ClassVar[RoleTypes] = RoleTypes.dms_architect
+    space: str
+    external_id: str = Field(alias="externalId")
+    version: str | None = Field(
```
Comment (nikokaoja) [inline cognite/neat/rules/models/_rules/dms_architect_rules.py:30]: Attach filed validator from information architect here.

File: cognite/neat/rules/models/_rules/dms_architect_rules.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        description="Data model version",
+        min_length=1,
+        max_length=43,
+    )
+
+    contributor: str | list[str] = Field(
```
Comment (nikokaoja) [inline cognite/neat/rules/models/_rules/dms_architect_rules.py:36]: Same thing for contributor, there is a method for turning string into list if there is comma

File: cognite/neat/rules/models/_rules/dms_architect_rules.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        updated: datetime | None = None,
+    ):
+        information = InformationMetadata.from_domain_expert_metadata(
+            metadata, None, None, version, contributor, created, updated
+        ).model_dump()
+
+        return cls.from_information_architect_metadata(information, space, external_id)
+
+
+class DMSProperty(SheetEntity):
```
Comment (nikokaoja) [inline cognite/neat/rules/models/_rules/dms_architect_rules.py:70]: Would it make sense that you inhering from `InformationProperty` and extend it with additional ?
There is already method `is_list` and `is_mandatory`?

File: cognite/neat/rules/models/_rules/dms_architect_rules.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        return value
+
+
+class DMSContainer(SheetEntity):
+    container: str = Field(alias="Container")
+    description: str | None = Field(None, alias="Description")
+    constraint: str | None = Field(None, alias="Constraint")
+
+
+class DMSView(SheetEntity):
```
Comment (nikokaoja) [inline cognite/neat/rules/models/_rules/dms_architect_rules.py:99]: You might want to do the same thing you did with the `Properties` sheet here, ie. renaming `Classes` sheet to `Views` and adding these additional columns

**Review by nikokaoja — APPROVED** [review]
LGTM

File: cognite/neat/rules/models/_rules/dms_architect_rules.py (hunk 1 above)
Comment (doctrino) [reply cognite/neat/rules/models/_rules/dms_architect_rules.py]: Moved out a type for it.

File: cognite/neat/rules/models/_rules/dms_architect_rules.py (hunk 2 above)
Comment (doctrino) [reply cognite/neat/rules/models/_rules/dms_architect_rules.py]: Moved out to special annotated type

File: cognite/neat/rules/models/_rules/dms_architect_rules.py (hunk 3 above)
Comment (doctrino) [reply cognite/neat/rules/models/_rules/dms_architect_rules.py:66]: Composition solution implemented.

**Pull request merged**